src/components/uiKit/Fences/Modal/index.tsx

```diff
@@ -17,16 +17,21 @@ const iconSize = {
   height: 45,
 };
 
-const AuthorizedStudiesConnectionModal = () => {
+const FencesConnectionModal = () => {
   const dispatch = useDispatch();
-  const { isFenceConnectionModalOpen, connectionStatus, fencesConnectError, loadingFences } =
+  const { modalConnectionParams, connectionStatus, fencesConnectError, loadingFences } =
     useFenceConnection();
 
   const isDcfConnected = connectionStatus[FENCE_NAMES.dcf] === FENCE_CONNECTION_STATUSES.connected;
   const isGen3Connected =
     connectionStatus[FENCE_NAMES.gen3] === FENCE_CONNECTION_STATUSES.connected;
 
-  const onClose = () => dispatch(fenceConnectionActions.toggleConnectionModal(false));
+  const onClose = () => {
+    if (modalConnectionParams.onClose) {
```

Comment:
1. suggestion (optional): `modalConnectionParams.onClose && modalConnectionParams.onClose()`
2. You should not have to explicitly pass down `undefined` . However, if the Types are a little bit messy, forget about it.

Reply:
the type are 'undefined' or a function, typescript will warms us if someone pass something else.